sdk/servicebus/service-bus/src/track2/models.ts

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ServiceBusMessage, DeadLetterOptions } from "../serviceBusMessage";
+import { TokenCredential } from "@azure/core-amqp";
+import { OperationOptions } from "@azure/core-auth";
+
+/**
+ * An opaque class, used internally to manage AMQP connections for sessions.
+ */
+// TODO: should probably implement Closeable so the user can just shut down all
+// connections when their app is done processing.
+export class SessionConnections {}
+
+/**
+ * Information needed to target a session.
+ */
+export interface Session {
+  /**
+   * The ID of a session to target or `undefined` to pick an session that is not currently
+   * owned.
+   */
+  id?: string;
+  /**
+   * A shared instance that allows multiple sessions to share the same underlying AMQP
+   * connection.
+   *
+   * A single instance should be created for your application and passed to each
+   * ReceiverClient you create that processes a session.
+   */
+  connections: SessionConnections;
+}
+
+export function isSession(possibleSession: Session | any): possibleSession is Session {
+  return (
+    (possibleSession as Session).connections &&
+    typeof (possibleSession as Session).connections === "object"
+  );
+}
+
+// TODO: make this an actual interface that's not just in terms of what's
+// on ServiceBusMessage. There's no reason to advertise that it's actually
+// a concrete class (or is there?)
+// the "action" methods here are on the context object instead.
+export type ReceivedMessage = Omit<
+  ServiceBusMessage,
+  | "complete"
+  | "abandon"
+  | "defer"
+  | "deadletter"
+  // Um..am I doing something odd here or is this a normal thing to exclude?
```

Comment:
I'm not sure to be honest...

Reply:
I think we want to move this from a concrete class to an interface underneath anyways so this probably won't be an issue going forward.